### PR TITLE
fix: merge the decorator data with the corresponding runtime symbol

### PR DESCRIPTION
**Pull request by vsavkin**
The PR merges the docs from decorator symbols and their corresponding runtime symbols (e.g., ContentChildren and ContentChildrenDecorator). The source code has to look this way because of the IDE support, but it doesn't make sense for the docs site because the users think of the two tokens as one.

cc @petebacondarwin @IgorMinar

### CODE REVIEW

File: tools/api-builder/angular.io-package/processors/addJadeDataDocsProcessor.js [hunk 1]
```diff
@@ -65,83 +124,60 @@ module.exports = function addJadeDataDocsProcessor() {
...
-              howToUse: howToUse,
-              whatItDoes: whatItDoes,
-              security: security
-            };
-
-            if (exportDoc.symbolTypeName) dataDoc.varType = titleCase(exportDoc.symbolTypeName);
-            if (exportDoc.originalModule) dataDoc.originalModule = exportDoc.originalModule;
-            return dataDoc;
           })
+          .filter(function(s) { return !!s; })
```
Comment (IgorMinar) [inline tools/api-builder/angular.io-package/processors/addJadeDataDocsProcessor.js]: please add a comment about what this does

**Comment by naomiblack** [comment]
could someone provide context about what this fix is?

**Comment by naomiblack** [comment]
this appears to break the dart docs generation with `gulp check-deploy`

error:
```
[13:06:04] Starting 'build-dart-cheatsheet'...
[Error: No provider for "mergeDecoratorDocs"! (Resolving: mergeDecoratorDocs)]
Error: No provider for "mergeDecoratorDocs"! (Resolving: mergeDecoratorDocs)
    at error (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:22:12)
    at Object.parent.get (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:9:13)
    at get (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:54:19)
    at /Users/naomib/github/angular.io/node_modules/di/lib/injector.js:71:14
    at Array.map (native)
    at invoke (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:70:31)
    at /Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:119:18
    at Array.forEach (native)
    at /Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:118:25
    at Array.forEach (native)
    at Dgeni.configureInjector (/Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:117:14)
    at Dgeni.generate (/Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:149:8)
    at buildDartCheatsheet (/Users/naomib/github/angular.io/gulpfile.js:1327:18)
    at Gulp.<anonymous> (/Users/naomib/github/angular.io/gulpfile.js:640:10)
    at module.exports (/Users/naomib/github/angular.io/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/naomib/github/angular.io/node_modules/orchestrator/index.js:273:3)
[13:06:08] 'build-dart-cheatsheet' errored after 4.09 s
[13:06:08] Error: No provider for "mergeDecoratorDocs"! (Resolving: mergeDecoratorDocs)
    at error (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:22:12)
    at Object.parent.get (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:9:13)
    at get (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:54:19)
    at /Users/naomib/github/angular.io/node_modules/di/lib/injector.js:71:14
    at Array.map (native)
    at invoke (/Users/naomib/github/angular.io/node_modules/di/lib/injector.js:70:31)
    at /Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:119:18
    at Array.forEach (native)
    at /Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:118:25
    at Array.forEach (native)
    at Dgeni.configureInjector (/Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:117:14)
    at Dgeni.generate (/Users/naomib/github/angular.io/node_modules/dgeni/lib/Dgeni.js:149:8)
    at buildDartCheatsheet (/Users/naomib/github/angular.io/gulpfile.js:1327:18)
    at Gulp.<anonymous> (/Users/naomib/github/angular.io/gulpfile.js:640:10)
    at module.exports (/Users/naomib/github/angular.io/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/naomib/github/angular.io/node_modules/orchestrator/index.js:273:3)
```

**Comment by naomiblack** [comment]
merged, live.

**Pull request closed without merging**